src/RamExistenceCheck.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2019, The Souffle Developers. All rights reserved
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+/************************************************************************
+ *
+ * @file RamExistenceCheck.h
```

Comment:
There is a RamExistenceCheck class and a RamExistenceCheckAnalysis class. RamExistenceCheckAnalysis  is in RamExistenceCheck.h and RamExistenceCheck is in RamOperation.h. This is confusing.

Could you change this to RamExistenceCheckAnalysis.h? (Which is now inconsistent with the other Ram.*Analysis classes so they should be refactored at some point, too)

Reply:
Good point thanks :+1: